streaming/src/main/scala/org/apache/spark/streaming/DStreamGraph.scala

```diff
@@ -160,6 +160,10 @@ final private[streaming] class DStreamGraph extends Serializable with Logging {
     }
   }
 
+  def getMaxInputStreamRememberDuration(): Duration = {
```

Comment:
I would add a doc here that explains that the max remember duration can be used to set a conservative threshold on when it is safe to remember older data. Otherwise it's not super obvious why callers are using this particular time.